### PR TITLE
Issue #1326 Enable hyperkit as driver for MacOS

### PR DESCRIPTION
**Pull request by praveenkumar**
Following steps need to test current POC.

- Get https://github.com/praveenkumar/docker-machine-driver-hyperkit and `make build`
- Make sure you disable volume mount check because with hyperkit mount driver created as `/mnt/vda1` instead `/mnt/sda1` 
     - `minishift config set skip-check-storage-mount true`

- Get the minishift binary for this PR
- `minishift start --vm-driver hyperkit`

### CODE REVIEW

**Comment by LalatenduMohanty** [comment]
@praveenkumar mentioned in IRC I can also reproduce following error

```
etting to WaitForSSH function...
(minishift) Calling .GetSSHHostname
(minishift) Calling .GetSSHPort
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHKeyPath
(minishift) Calling .GetSSHUsername
Using SSH client type: external
Using SSH private key: /Users/lmohanty/.minishift/machines/minishift/id_rsa (-rw-------)
&{[-F /dev/null -o PasswordAuthentication=no -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=quiet -o ConnectionAttempts=3 -o ConnectTimeout=10 -o ControlMaster=no -o ControlPath=none docker@192.168.64.13 -o IdentitiesOnly=yes -i /Users/lmohanty/.minishift/machines/minishift/id_rsa -p 22] /usr/bin/ssh <nil>}
About to run SSH command:
exit 0
SSH cmd err, output: exit status 255:
Error getting ssh command 'exit 0' : ssh command error:
command : exit 0
err     : exit status 255
output  :
. FAIL E0228 17:36:08.567253   53815 start.go:369] Error starting the VM: Error creating the VM. Error creating machine: Error detecting OS: Too many retries waiting for SSH to be available.  Last error: Maximum number of retries (60) exceeded. Retrying.
Error starting the VM: Error creating the VM. Error creating machine: Error detecting OS: Too many retries waiting for SSH to be available.  Last error: Maximum number of retries (60) exceeded
```

**Comment by praveenkumar** [comment]
@LalatenduMohanty updated the documents.

File: cmd/minishift/cmd/start_preflight.go [hunk 1]
```diff
@@ -41,8 +41,8 @@ import (
 	stringUtils "github.com/minishift/minishift/pkg/util/strings"
 )
 
-const (
-	StorageDisk = "/mnt/sda1"
+var (
+	storageDisk = []string{"/mnt/sda1", "/mnt/vda1"}
```
Comment (praveenkumar) [inline cmd/minishift/cmd/start_preflight.go]: @gbraad this is what I am planning as of now to handle disk checks.

**Comment by robin-owen** [comment]
Just reviewed the updated documentation. Looks good to me! :+1:

**Comment by praveenkumar** [comment]
> As a side note, we'll also need to update the information in the troubleshooting section at some point. :)

If we find some issue around hyperkit then yes we will update the troubleshooting section also but as of now, it will go as it is and let's hear from users if they face any issue around it.

**Comment by praveenkumar** [comment]
As of now this depends on #2097 PR since we will not be going to add check drive as the list instead use `?da`

**Comment by LalatenduMohanty** [comment]
@praveenkumar #2097 has been merged now, so please update this PR.

**Comment by praveenkumar** [comment]
@LalatenduMohanty update PR now.

**Comment by praveenkumar** [comment]
retest this please.

File: docs/source/getting-started/setting-up-driver-plugin.adoc [hunk 2]
```diff
@@ -183,6 +183,52 @@ $ sudo chmod u+s /usr/local/bin/docker-machine-driver-xhyve
...
 ====
 
+[[hyperkit-driver-install]]
+== hyperkit Driver
+
+{project} is currently tested against *docker-machine-driver-hyperkit* version 1.0.0.
+
+
+To install the hyperkit driver, you need to download and install the *docker-machine-driver-hyperkit* binary and place it in a directory which is on your `PATH` environment variable.
+The directory *_/usr/local/bin_* is most likely a good choice, since it is the default installation directory for Docker Machine binaries.
```
Comment (LalatenduMohanty) [inline docs/source/getting-started/setting-up-driver-plugin.adoc]: s/most likely a good choice/is a good choice/

Comment (LalatenduMohanty) [inline docs/source/getting-started/setting-up-driver-plugin.adoc]: s/most likely a good choice/is a good choice/

Comment (LalatenduMohanty) [reply docs/source/getting-started/setting-up-driver-plugin.adoc]: Fixed it locally and merged it.

**Comment by LalatenduMohanty** [comment]
Merged https://github.com/minishift/minishift/commit/d314d539836105846e3963687a99c13b0def41a8

**Pull request closed without merging**